internal/layer2/announcer.go

```diff
@@ -181,6 +212,8 @@ func (a *Announce) shouldAnnounce(ip net.IP) dropReason {
 
 // SetBalancer adds ip to the set of announced addresses.
 func (a *Announce) SetBalancer(name string, ip net.IP) {
+	// Call spamStart at the end of the function without holding the lock
+	defer a.spamStart(ip)
```

Comment:
How about the following?

```go
	defer func() {
		a.spamCh <- ip
	}()
```

We can then get rid of `spamStart()`. In any case, if we keep this function I'd rather name it `doSpam()` since having "start" in the name implies that there is also "stop", which isn't the case.

Reply:
I went with `doSpam()`, just a matter of taste, but if other people prefer the anonymous func i'll change it